src/redisearch_rs/inverted_index/src/lib.rs

```diff
@@ -677,5 +677,39 @@ impl<E: Encoder> InvertedIndex<E> {
     }
 }
 
+/// A reader that filters out records that do not match a given field mask. It is used to
+/// filter records in an index based on their field mask, allowing only those that match the
+/// specified mask to be returned.
+pub struct FilterMaskReader<I> {
+    /// Mask which a record needs to match to be valid
+    mask: t_fieldMask,
+
+    /// The inner reader that will be used to read the records from the index.
+    inner: I,
+}
+
+impl<I: Iterator<Item = RSIndexResult>> FilterMaskReader<I> {
```

Comment:
At some point, maybe not now, maybe it could make sense to have a result reader trait bound to `Iterator<Item = RSIndexResult>`?

Reply:
Do you mean a new trait that has `Iterator<Item = RSIndexResult>` as the super trait? Then yes, I think it will match 1-to-1 with the new iterator redis made.

Reply:
yes exactly.